src/tests/ftest/util/command_utils.py

```diff
@@ -570,6 +570,28 @@ def run(self, raise_exception=None):
                 "<{}> command failed: {}".format(self.command, error)) from error
         return self.result
 
+    def set_command(self, sub_command_list=None, **kwargs):
+        """Set the command and its arguments.
+
+        Args:
+            sub_command_list (list, optional): a list of sub commands used to
+                define the command to execute. Defaults to None, which will run
+                the command as it is currently defined.
+
+        Raises:
+            CommandFailure: if an unknown parameter is provided for the full command
+
+        """
+        # Set up the full command by setting each specified sub-command
+        full_command = self
+        if sub_command_list is not None:
+            for sub_command in sub_command_list:
+                full_command.set_sub_command(sub_command)
+                full_command = full_command.sub_command_class
+
+        # Update any argument values for the full command
+        self.update_params(**kwargs)
```

Comment:
I like this. One implicit improvement to using `update_params` is that it always displays the updated value

Reply:
In this case it's the handling of an unexpected argument